docs/data/base/guides/interoperability-with-tailwind-css/interoperability-with-tailwind-css.md

```diff
@@ -0,0 +1,451 @@
+  ref: React.ForwardedRef<HTMLSpanElement>,
+){
+  return (<SliderUnstyled
+    {...props}
+    ref={ref}
+    componentsProps={{
+      thumb: { className: "ring-cyan-500 dark:ring-cyan-400 ring-2 w-4 h-4 -mt-1 -ml-2 flex items-center justify-center bg-white rounded-full shadow absolute" }
+      root: { className: "w-full relative inline-block h-2 cursor-pointer" },
+      rail: { className: "bg-slate-100 dark:bg-slate-700 h-2 w-full rounded-full block absolute" },
+      track: { className: "bg-cyan-500 dark:bg-cyan-400 h-2 absolute rounded-full" }
+    }}
+  />);
+});
+
+export default Slider;
+```
+
+You will notice that the `componentsProps` above, defines the class names for the different parts of the slider.
+Most of the classes are copied from the initial markup, with small adjustments in order to make the `Slider` functional.
+Next, let's use this `Slider` in the `Player` component, by replacing the lines 24-29 with:
+
+```diff
+--- a/src/App.tsx
++++ b/src/App.tsx
+@@ -1,4 +1,5 @@
+ import * as React from 'react';
++import Slider from './Slider';
+
+ const Player = React.forwardRef(function Player(props: { className?: string }, ref: React.ForwardedRef<HTMLDivElement>) {
+   const { className, ...other } = props;
+@@ -21,12 +22,7 @@ const Player = React.forwardRef(function Player(props: { className?: string }, r
+         </div>
+         <div className="space-y-2">
+           <div className="relative">
+-            <div className="bg-slate-100 dark:bg-slate-700 rounded-full overflow-hidden">
+-              <div className="bg-cyan-500 dark:bg-cyan-400 w-1/2 h-2" role="progressbar" aria-label="music progress" aria-valuenow={1456} aria-valuemin={0} aria-valuemax={4550}></div>
+-            </div>
+-            <div className="ring-cyan-500 dark:ring-cyan-400 ring-2 absolute left-1/2 top-1/2 w-4 h-4 -mt-2 -ml-2 flex items-center justify-center bg-white rounded-full shadow">
+-              <div className="w-1.5 h-1.5 bg-cyan-500 dark:bg-cyan-400 rounded-full ring-1 ring-inset ring-slate-900/5"></div>
+-            </div>
++            <Slider step={50} defaultValue={1456} max={4550} min={0} />
+           </div>
+           <div className="flex justify-between text-sm leading-6 font-medium tabular-nums">
+             <div className="text-cyan-500 dark:text-slate-100">24:16</div>
+```
+
+After this you will get the following result:
+
+<img src="/static/images/base-with-tailwind-css/player-slider.png" alt="Example UI using Tailwind CSS - player" width="638" />
+
+You will notice now that the slider is interactive, however it doesn't look exactly as the initial picture.
+This is because we haven't defined the element that represents the dot inside the thumb.
+In order to do this, is not enough to just propagate classes for the thumb, we need to actually render a custom component for it.
+We can do this easily by passing the custom component in the `components` props of the `Slider`.
+This is what we need to do
```

Comment:
```suggestion
Even though the slider is now interactive, it still does not look exactly like the original design.
This is because we haven't defined the element that represents the dot inside the thumb.

To do this, it's not enough to just use classes for the thumb, we need also to render a custom component that gets passed in the `components` prop of the `Slider`.
```